python/ray/data/tests/conftest.py

```diff
@@ -14,12 +14,18 @@
 def aws_credentials():
     import os
 
+    # Credentials dict that can be passed as kwargs to pa.fs.S3FileSystem
+    credentials = dict(
+        access_key="testing", secret_key="testing", session_token="testing"
+    )
+
     old_env = os.environ
-    os.environ["AWS_ACCESS_KEY_ID"] = "testing"
-    os.environ["AWS_SECRET_ACCESS_KEY"] = "testing"
+    os.environ["AWS_ACCESS_KEY_ID"] = credentials["access_key"]
+    os.environ["AWS_SECRET_ACCESS_KEY"] = credentials["secret_key"]
     os.environ["AWS_SECURITY_TOKEN"] = "testing"
-    os.environ["AWS_SESSION_TOKEN"] = "testing"
-    yield
+    os.environ["AWS_SESSION_TOKEN"] = credentials["session_token"]
+
+    yield credentials
```

Comment:
@clarkzinzow I ran into problems on the CI machines with pyarrow 6.0.1 (but not with 4.0.1) and manual debugging suggested that there were credential issues. Calling `fs.create_dir("tmp/test")` multiple times would result in 403 errors from `moto_server`. Passing the credentials as kwargs to `pa.fs.S3FileSystem` solved the problem for me.

I'm not sure how this comes about, and I couldn't reproduce it locally, but since this is a testing setup I think this adjustment should be fine.

I found the same error in your open PRs updating to PyArrow 6.0.1 so this might be helpful.

Reply:
what's the difference between the two versions? Do we want to update the version in this PR?

Reply:
It's only not working on Buildkite instances fwiw, I've debugged in a repro instance. Happy to hop on a call to show you how to debug this efficiently